yarn-project/simulator/src/public/side_effect_trace.ts

```diff
@@ -81,6 +83,36 @@ export class PublicSideEffectTrace implements PublicSideEffectTraceInterface {
     this.sideEffectCounter++;
   }
 
+  // This tracing function gets called everytime we start simulation/execution.
+  // This happens both when starting a new top-level trace and the start of every nested trace
+  // We use this to collect the AvmContractBytecodeHints
+  public traceExecutionStart(
```

Comment:
Thoughts on renaming this something like `traceGetBytecode`, `traceBytecodeRetrieval`, `traceGetContract` or `traceContractRetrieval`?

Reply:
yeah I'm fine with `traceGetBytecode`, i wouldn't pick anything with contract in it to avoid confusing it with the `traceContractInstance`